player/tests/test.rs

```diff
@@ -0,0 +1,169 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*! Tester for WebGPU
+ *  It enumerates the available backends on the system,
+ *  and run the tests through them.
+ *
+ *  Test requirements:
+ *    - all IDs have the backend `Empty`
+ *    - all expected buffers have `MAP_READ` usage
+ *    - last action is `Submit`
+!*/
+
+use player::{gfx_select, GlobalPlay, IdentityPassThroughFactory};
+use std::{
+    fs::{read_to_string, File},
+    path::{Path, PathBuf},
+    ptr, slice,
+};
+
+#[derive(serde::Deserialize)]
```

Comment:
Would it make sense to import this trait and use it without the prefix?

Reply:
we totally could, but it's unlikely to substantially reduce the code size in this case, since we only use this a few times